pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go

```diff
@@ -96,19 +98,39 @@ func (mutator *VMsMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1.
 }
 
 func (mutator *VMsMutator) setDefaultMachineType(vm *v1.VirtualMachine) {
+	// Nothing to do, let's the validating webhook fail later
 	if vm.Spec.Template == nil {
-		// nothing to do, let's the validating webhook fail later
 		return
 	}
-	machineType := mutator.ClusterConfig.GetMachineType()
 
-	if machine := vm.Spec.Template.Spec.Domain.Machine; machine != nil {
-		if machine.Type == "" {
-			machine.Type = machineType
-		}
-	} else {
-		vm.Spec.Template.Spec.Domain.Machine = &v1.Machine{Type: machineType}
+	if machine := vm.Spec.Template.Spec.Domain.Machine; machine != nil && machine.Type != "" {
+		return
+	}
+
+	if vm.Spec.Template.Spec.Domain.Machine == nil {
+		vm.Spec.Template.Spec.Domain.Machine = &v1.Machine{}
+	}
+
+	if vm.Spec.Preference != nil {
```

Comment:
No need to check for `nil` here, because `mutator.InstancetypeMethods.FindPreferenceSpec(vm)` will return nil if preference is not assigned to the VM.